Assets/SEE/Utils/VersionControlSystems.cs

```diff
@@ -0,0 +1,144 @@
+            }
+            else
+            {
+                // File got added and is only available in the other commit.
+                return "";
+            }
+        }
+
+        /// <summary>
+        /// Returns the source code from the revision, that gets compared.
+        /// </summary>
+        /// <returns>the source code from the revision, that gets compared.</returns>
+        public string ShowOriginal(string repositoryPath, string fileName, string oldCommitID, string newCommitID)
+        {
+            Repository repo = new(repositoryPath);
+            Commit newCommit = repo.Lookup<Commit>(newCommitID);
+            Commit oldCommit = repo.Lookup<Commit>(oldCommitID);
+
+            CompareOptions compareOptions = new()
+            {
+                Algorithm = DiffAlgorithm.Myers,
+                Similarity = new SimilarityOptions
+                {
+                    RenameDetectionMode = RenameDetectionMode.Default,
+                }
+            };
+
+            // Compare the commits
+            var changes = repo.Diff.Compare<TreeChanges>(oldCommit.Tree, newCommit.Tree, compareOptions)
+                .Where(change => (change.Path == fileName || change.OldPath == fileName));
+
+            foreach (TreeEntryChanges change in changes)
+            {
+                // Case: file got renamed.
+                if ((change.Status == ChangeKind.Renamed || change.Status == ChangeKind.Copied) &&
+                    (change.Path == fileName || change.OldPath == fileName))
+                {
+                    // Change the filename in the codewindow.
+                    name = change.Path;
+                    // Get renamed file from commit.
+                    var renamedBlob = newCommit[change.Path]?.Target as Blob;
+                    if (renamedBlob != null)
+                    {
+                        return renamedBlob.GetContentText();
+                    }
+                }
+                // Case: File got deleted.
+                if (change.Status == ChangeKind.Deleted && change.Path == fileName)
+                {
+                    // Mark filename in the codewindow as deleted.
+                    name = fileName;
+                    return "";
+                }
+            }
+            Blob blob = newCommit[fileName]?.Target as Blob;
+            return blob.GetContentText();
+        }
+
+        /// <summary>
+        /// Returns the updated filename.
+        /// When a renaming took place, the new name gets returned.
+        /// When a file got deleted, the original filename gets returned.
+        /// Otherwise null gets returned.
+        /// </summary>
+        /// <returns>the updated filename</returns>
+        public string ShowName()
+        {
+            return name;
+        }
+    }
+        /// <summary>
+        /// Implements the functionality of IVersionControl for SVN.
+        /// </summary>
+        public class SvnVersionControl : IVersionControl
```

Comment:
I wonder why these classes implementing `IVersionControl` need to have state. Couldn't they be `static` instead? Just a thought.

Reply:
Compiler Error CS0714
'static type' : static classes cannot implement interfaces

Interfaces may define non-static methods on objects and hence may not be implemented by static classes. To resolve this error, make sure your class does not attempt to implement any interfaces.

Reply:
Yes, but we could apply the singleon pattern. Anyhow, maybe we leave it this way.

Reply:
Why is `GitVersionControl `nested in a `MonoBehaviour `(`VersionControlSystems`)?

I fixed it.